marker_rustc_driver/src/main.rs

```diff
@@ -71,6 +95,49 @@ impl rustc_driver::Callbacks for MarkerCallback {
     }
 }
 
+fn register_tracked_env(sess: &mut rustc_session::parse::ParseSess, vars: &[(&'static str, String)]) {
+    use rustc_span::Symbol;
+    let env = sess.env_depinfo.get_mut();
+
+    for (key, value) in vars {
+        env.insert((Symbol::intern(key), Some(Symbol::intern(value))));
+    }
+}
+
+fn register_tracked_files(sess: &mut rustc_session::parse::ParseSess, lint_crates: &[LintCrateInfo]) {
+    use rustc_span::Symbol;
+
+    let files = sess.file_depinfo.get_mut();
+
+    // Cargo sets the current directory, to the folder containing the `Cargo.toml`
+    // file of the compiled crate. Therefore, we can use the relative path.
+    let cargo_file = Path::new("./Cargo.toml");
+    if cargo_file.exists() {
+        files.insert(Symbol::intern("./Cargo.toml"));
+    }
+
+    // At this point, the lint crate paths should be absolute. Therefore, we
+    // can just throw it in the `files` set.
+    for lint_crate in lint_crates {
+        if let Some(name) = lint_crate.path.to_str() {
+            files.insert(Symbol::intern(name));
+        } else {
+            // The name is not a valid UTF-8 String. This is not ideal but at the
+            // same time not problematic enough to throw an error. It just means
+            // that the lint crate can't be tracked and the driver might reuse
+            // a cache if only that crate was changed.
+        }
+    }
+
+    // Track the driver executable in debug builds
+    #[cfg(debug_assertions)]
+    if let Ok(current_exe) = env::current_exe()
```

Comment:
`if let Ok(_)` patterns always catch my eye, because they silently ignore an error. I'd rather log the error if it happens. But, for that there should be `tracing` installed

Reply:
I can use unwrap, instead, either way, I think this is fine, since this is only enabled for debug builds.

Reply:
I'll change it to a match